### DISCUSSION TITLE
Training using nongeoreferenced Airbus Ship detection data

### DISCUSSION THREAD

**Discussion by julled** [azavea/raster-vision#1592] [Q&A]
Hi, i just found your very nice framework. Its nice that its specifically tailored for satellite images.

I am wondering how you recommend the usage of the framework for the usage on non-geotiff / nongeoreferenced / png data like the Airbus Ship detection data. We want to use the data in a object detection task. 
We converted the segmentation labels to a coco json label file with one rectangle per boat instance.

I already found out that its possible to load the pngs using a `rasteriosource`. 

What is now unclear to me is how to feed in the coco boat rectangle labels for usage in a `ObjectDetection` pipeline. One option is to convert the coco jsons to geojsons. Is it possible to use geojson files with labels in pixel coordinates in combination with the  `rasteriosource`? 

Or do you recommend any other format ?

Thank you for your help!

**Comment by AdeelH** [comment]
Hi, I suggest using the `master` branch.

You do not need to use `RasterioSource` and you do not need to convert annotations to GeoJSON; RV can handle pre-chipped data. You will need to create an [ObjectDetectionImageDataset](https://raster-vision--1594.org.readthedocs.build/en/1594/api_reference/_generated/rastervision.pytorch_learner.dataset.object_detection_dataset.ObjectDetectionImageDataset.html). This requires your annotations to be a JSON file in the [COCO format](https://opencv.org/introduction-to-the-coco-dataset/#:~:text=COCO%20stores%20data%20in%20a,%2C%20testing%2C%20and%20validation%20purposes.).

Once you have the dataset, you can proceed with training as normal. See [this tutorial on training](https://raster-vision--1594.org.readthedocs.build/en/1594/usage/tutorials/train.html). Note where `train_ds` and `val_ds` are being defined. You will need to change these to be `ObjectDetectionImageDataset`s. You will also need to replace the semantic segmentation-specific classes with equivalent object detection ones.

Also see this [discussion](https://github.com/azavea/raster-vision/discussions/1585#discussioncomment-4320050).

Let us know if you run into any trouble.

**Comment by julled** [comment]
Thanks for the detailed answer, this helps alot already!

Our usecase would be now to use the Airbus dataset as a base source for data and after training do a finetune on a smaller custom dataset. Is finetuning possible using rastervision?

**Reply by lewfish** [reply]
Yes, you can finetune. See https://docs.rastervision.io/en/latest/framework/misc.html#transfer-learning-using-models-trained-by-rv

**Comment by AdeelH** [comment]
@julled 

While preparing [this notebook](https://raster-vision--1602.org.readthedocs.build/en/1602/usage/tutorials/prechipped_datasets.html), I realized that although the Airbus aircraft detection dataset has non-GeoTIFF images, those images are still large enough (2K+ by 2K+ pixels) to require chipping.

If this is also true for the Airbus dataset you are using, then, yes, you should be using `RasterioSource` and not the `ImageDataset` as I said in my earlier comment.

I can try to provide more guidance if you share some more details about your dataset.
